### PR TITLE
Update Java Destination Docs

### PR DESCRIPTION
**Pull request by MatKuhr**
This PR updates the Java docs on destinations to explain how to register custom destinations in productive code.

### CODE REVIEW

**Comment by Emdee89** [comment]
Looks good, but has to wait until the release after 3.32.0 AFAIK.

**Comment by MatKuhr** [comment]
> Looks good, but has to wait until the release after 3.32.0 AFAIK.

In the past we shipped documentation updates without a tight coupling to the release schedule. Since this will go GA today/tomorrow I would like to ship the docs update now.

File: docs/java/features/connectivity/destinations.mdx [hunk 1]
```diff
@@ -75,10 +75,13 @@ destination.asHttp().decorate(DefaultErpHttpDestination::new);
 
 This ensures the mapping of all S/4 properties like `sap-client` and `sap-locale` as HTTP request headers.
 
-## Testing destinations in local environment ##
+## Registering Destinations at Runtime ##
 
-For testing purposes, the SDK provides functionality to provide such a destination in a local development environment. 
-This is especially useful when working with destinations from the destination service since the service is not available in a local setting.
+The SDK offers convenient ways to create a destination at runtime and register it so that it will be available via `DestinationAccessor.getDestination()`.
+This is especially useful when working in a local environment.
```
Comment (artemkovalyov) [inline docs/java/features/connectivity/destinations.mdx:81]:
```suggestion
:::tip
This is especially useful when working in a local environment.
:::
```

**Review by artemkovalyov — APPROVED** [review]
Good update. I think we can merge it even  before release.

**Pull request merged**